### PR TITLE
Bug 666066: Show warnings for obfuscated/minified code.

### PR DESCRIPTION
**Pull request by kmaglione**

### CODE REVIEW

**Comment by kmaglione** [comment]
Looks like this needs #275 to build.

**Comment by magopian** [comment]
@kmaglione could you please rebase?

**Comment by kmaglione** [comment]
I rebased on top of #278, since it seemed likely there would be conflicts.

File: validator/testcases/javascript/traverser.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 IGNORE_POLLUTION = False
 POLLUTION_COMPONENTS_PATH = re.compile(r"/?components/.*\.jsm?")
 POLLUTION_EXCEPTIONS = set(["Cc", "Ci", "Cu", ])
+MAX_LINE_COMPLEXITY = 100
```
Comment (magopian) [inline validator/testcases/javascript/traverser.py]: That looks like a lot. I was expecting something like 10. Are you afraid of false positives?

File: validator/testcases/javascript/traverser.py [hunk 2]
```diff
@@ -83,6 +85,22 @@ def run(self, data):
             raise
         self._debug("END>>")
 
+        for line in sorted(self.complexity.keys()):
```
Comment (magopian) [inline validator/testcases/javascript/traverser.py]: Why sorting?

**Comment by magopian** [comment]
There's nothing in there to detect code obfuscation, right? Also, curious why you didn't go with the "count number of `;` per line" method that seemed easier.

It's missing tests.

File: validator/testcases/javascript/traverser.py (hunk 1 above)
Comment (magopian) [inline validator/testcases/javascript/traverser.py]: Following in-person discussion, seems that 100 is just right.

File: validator/testcases/javascript/traverser.py (hunk 2 above)
Comment (magopian) [inline validator/testcases/javascript/traverser.py]: After in-person discussion, it's so the warnings are displayed in the correct order (the key is the line number)

**Comment by magopian** [comment]
As discussed in-person: needs tests, and then it's good to merge

r+wc

File: validator/testcases/javascript/traverser.py (hunk 1 above)
Comment (muffinresearch) [inline validator/testcases/javascript/traverser.py]: @magopian Can you share the rationale here too so it's documented? We might want similar checks in the JS validator.

Comment (magopian) [inline validator/testcases/javascript/traverser.py]: Oh, wow, I should have noted that at the time, I completely forgot :'( @kmaglione could you please explain it again?

**Comment by EnTeQuAk** [comment]
I'm going to close this for cleaning up reasons, not sure how relevant this PR still is. Feel free to reopen anytime though :)

**Pull request closed without merging**